health-check/src/pg.ts

```diff
@@ -0,0 +1,63 @@
+import { ServerResponse, createServer } from 'http';
+import * as dotenv from 'dotenv';
+import { exec } from 'child_process';
+
+dotenv.config();
+
+function checkIsPostgresReady() {
+  const command = `pg_isready -h ${process.env.POSTGRES_HOST} -p ${process.env.POSTGRES_PORT}`;
+  return new Promise((resolve, reject) => {
+    exec(command, (err, stdout) => {
+      if (err) {
+        reject(err);
+        return;
+      }
+      const ready = stdout.includes('accepting connections');
+      resolve(ready);
+    });
+  });
+}
+
+async function checkPostgresHealth(res: ServerResponse) {
+  try {
+    const isReady = await checkIsPostgresReady();
+
+    if (isReady) {
+      res.writeHead(200, { 'Content-Type': 'text/plain' });
+      res.end('OK');
+    } else {
+      res.writeHead(500, { 'Content-Type': 'text/plain' });
+      res.end('Internal Server Error');
+    }
+  } catch (error) {
+    console.error(error);
+    res.writeHead(500, { 'Content-Type': 'text/plain' });
+    res.end('Internal Server Error');
+  }
+}
+
+const server = createServer(async (req, res) => {
+  const authHeader = req.headers.authorization;
+
+  if (authHeader && authHeader.startsWith('Bearer ')) {
+    const token = authHeader.split('Bearer ')[1];
+
+    if (token === process.env.SECRET) {
```

Comment:
This is very bad and easy to guess. Never check secrets this way, you need to use constant-time comparison.

https://en.wikipedia.org/wiki/Timing_attack

Reply:
I would not use this approach for another service, but this is a pretty basic service and I just wanted to add basic protection

Reply:
I mean if you add secret check, add it properly. I don't recall if JS already has built-in function for that, but there are definitely some packages that do that.

Reply:
this can be done natively using `crypto.timingSafeEqual`, I will submit a PR